tests/functional/data/appimage/create.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -e
+
+main() {
+    # Create the final AppImage
+    local arch="$1"
+    local tools_dir="$2"
+    local tmp_dir="$3"
+    local app_name="$4"
+    local app_id="org.foo.bar"
+    local app_dir="${tmp_dir}/dist/AppRun"
+    local output="${tmp_dir}/dist/${app_name}-${arch}.AppImage"
+
+    echo ">>> Adjusting file names to fit in the AppImage"
+    [ -d "${app_dir}" ] && rm -rf "${app_dir}"
+    mv -v "${tmp_dir}/dist/${app_name}" "${app_dir}"
+    mv -v "${app_dir}/${app_name}" "${app_dir}/AppRun"
+
+    echo ">>> Copying icons"
+    cp -v "${tools_dir}/DirIcon.png" "${app_dir}/.DirIcon"
+    cp -v "${tools_dir}/AppIcon.svg" "${app_dir}/${app_name}.svg"
+
+    echo ">>> Copying metadata files"
+    mkdir -pv "${app_dir}/usr/share/metainfo"
+    cp -v "${tools_dir}/${app_id}.appdata.xml" "${app_dir}/usr/share/metainfo"
+    mkdir -pv "${app_dir}/usr/share/applications"
+    cp -v "${tools_dir}/${app_id}.desktop" "${app_dir}/usr/share/applications"
+    ln -srv "${app_dir}/usr/share/applications/${app_id}.desktop" "${app_dir}/${app_id}.desktop"
+
+    echo ">>> Creating the AppImage file"
+    ${tools_dir}/appimagetool-${arch}.AppImage --no-appstream "${app_dir}" "${output}"
+
```

Comment:
Is there some reason for not implementing this shell-script as part of the test-case?

Reply:
Not really. I was thinking about possible future new AppImage tests, it would have been less duplicated code. I can move it into the test-case, let's think about it when needed :)